From 50c7ac12ae762b26c0dfdfc04cab9f25a1d2ca00 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:24 -0400
Subject: [block] blk-mq: add async parameter to blk_mq_start_stopped_hw_queues

Message-id: <1402670238-13916-37-git-send-email-snitzer@redhat.com>
Patchwork-id: 83800
O-Subject: [RHEL7.1 PATCH 036/150] blk-mq: add async parameter to blk_mq_start_stopped_hw_queues
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 1b4a325858f695a9b5041313602d34b36f463724
Author: Christoph Hellwig <hch@lst.de>
Date:   Wed Apr 16 09:44:54 2014 +0200

    blk-mq: add async parameter to blk_mq_start_stopped_hw_queues

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 963a821..da38088 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -700,7 +700,7 @@ void blk_mq_start_hw_queue(struct blk_mq_hw_ctx *hctx)
 }
 EXPORT_SYMBOL(blk_mq_start_hw_queue);
 
-void blk_mq_start_stopped_hw_queues(struct request_queue *q)
+void blk_mq_start_stopped_hw_queues(struct request_queue *q, bool async)
 {
  struct blk_mq_hw_ctx *hctx;
  int i;
@@ -711,7 +711,7 @@ void blk_mq_start_stopped_hw_queues(struct request_queue *q)
 
   clear_bit(BLK_MQ_S_STOPPED, &hctx->state);
   preempt_disable();
-  blk_mq_run_hw_queue(hctx, true);
+  blk_mq_run_hw_queue(hctx, async);
   preempt_enable();
  }
 }
diff --git a/include/linux/blk-mq.h b/include/linux/blk-mq.h
index 7588526..147503b 100644
--- a/include/linux/blk-mq.h
+++ b/include/linux/blk-mq.h
@@ -157,7 +157,7 @@ void blk_mq_complete_request(struct request *rq);
 void blk_mq_stop_hw_queue(struct blk_mq_hw_ctx *hctx);
 void blk_mq_start_hw_queue(struct blk_mq_hw_ctx *hctx);
 void blk_mq_stop_hw_queues(struct request_queue *q);
-void blk_mq_start_stopped_hw_queues(struct request_queue *q);
+void blk_mq_start_stopped_hw_queues(struct request_queue *q, bool async);
 
 /*
  * Driver command data is immediately after the request. So subtract request
-- 
1.7.1